### PR TITLE
Move connect api to rmt config

### PR DESCRIPTION
**Pull request by ngetahun**
## Description

This PR moves the scc api endpoint to the config file.

## Change Type

*Please select the correct option.*

- [ ] **Bug Fix** (a non-breaking change which fixes an issue)
- [ ] **New Feature** (a non-breaking change which adds new functionality)
- [ ] **Documentation Update** (a change which only updates documentation)

## Checklist

*Please check off each item if the requirement is met.*

- [x] I have verified that my code follows RMT's coding standards with `rubocop`.
- [x] I have reviewed my own code and believe that it's ready for an external review.
- [ ] I have provided comments for any hard-to-understand code.
- [ ] I have documented the `MANUAL.md` file with any changes to the user experience.
- [ ] RMT's test coverage remains at 100%.
- [ ] If my changes are non-trivial, I have added a changelog entry to notify users at `package/obs/rmt-server.changes`.

## Other Notes

Please use this space to provide notes or thoughts to the team, such as tips on how to review/demo your changes.

### CODE REVIEW

**Comment by ngetahun** [comment]
[[RMT] Move constants to config files for flexible setups](https://trello.com/c/563Oy522/2100-rmt-move-constants-to-config-files-for-flexible-setups)

**Comment by suse-tests-pass** [comment]
Well Done! Your tests are still passing.
https://ci.suse.de/job/scc-RMT-integration-tests/213045/console
If the given link has expired,you can force a Prophet rerun by just deleting this comment. (Merged f95c19658e9d95ceeeb2a7ab7d0a8cb3a04ec997 into d3226e0749bc27e5d424a626428115914e90d4d6)

**Comment by suse-tests-pass** [comment]
Well Done! Your tests are still passing.
https://ci.suse.de/job/scc-RMT-integration-tests/213500/console
If the given link has expired,you can force a Prophet rerun by just deleting this comment. (Merged cfb8971c770cae49afdc33d074400142983036d5 into d3226e0749bc27e5d424a626428115914e90d4d6)

File: lib/suse/connect/api.rb [hunk 1]
```diff
@@ -16,7 +16,15 @@ def initialize(response)
         end
       end
 
-      CONNECT_API_URL = 'https://scc.suse.com/connect'.freeze
+      def connect_api
+        uri_string = Settings.try(:scc).try(:host) || 'https://scc.suse.com/connect'
+        unless URI::DEFAULT_PARSER.make_regexp(['http', 'https', 'localhost']).match?(uri_string)
```
Comment (felixsch) [inline lib/suse/connect/api.rb:21]: So I managed to come around this check by using this string: `htxxtp://xxxxxxlocalhost:3000` (I swear this is the first try to trigger the exception!). I wonder, do we really need to validate the URI for sanity here?

If you run it with a malformed URI like the above you get:

```
SCC API request failed. Error details:
Request URL: htxxtp://xxxxxxlocalhost:3000/organizations/products?page=1
Response code: 0
Response body:
```

Isn't this enough for the user to find out the URI is just malformed? I'm undecided but would drop the check do make the code more easy and remove complexity. What do you think?

**Review by felixsch — CHANGES_REQUESTED** [review]

File: lib/suse/connect/api.rb (hunk 1 above)
Comment (ngetahun) [reply lib/suse/connect/api.rb:21]: We need to have the sanity checks. It's a cleaner way of handling the URIs. That url should've failed, I will check why it didn't.

Comment (ngetahun) [reply lib/suse/connect/api.rb]: Turns out the `localhost` regexp matcher will accept any string, including the malformed one you just caught. I have removed it and tests are now passing.

**Comment by suse-tests-pass** [comment]
Well Done! Your tests are still passing.
https://ci.suse.de/job/scc-RMT-integration-tests/213662/console
If the given link has expired,you can force a Prophet rerun by just deleting this comment. (Merged d2258f3e3cda130e627d95600f955e10ce32a834 into d3226e0749bc27e5d424a626428115914e90d4d6)

**Review by felixsch — APPROVED** [review]
lgtm :+1:

**How I reviewed this pull request:**

- I run the branch and checked the default configuration works out of the box
- Set the `scc » host` configuration to my localhost an check `sync` is working.
- Set to a malformed string and see the exception.
- Checked if there is no configuration at all fallback works

PS: Looks good, the only nitpicking thing is the exception is not handled and therefore the output is not really nicely formated:

```
I, [2021-09-22T11:05:41.825992 #6995]  INFO -- : Downloading data from SCC
I, [2021-09-22T11:05:41.826055 #6995]  INFO -- : Updating products
Traceback (most recent call last):
	13: from bin/rmt-cli:41:in `<main>'
	12: from /home/felixsch/.rvm/gems/ruby-2.5.8/gems/thor-1.1.0/lib/thor/base.rb:485:in `start'
	11: from /files/suse/rmt/lib/rmt/cli/base.rb:29:in `dispatch'
	10: from /files/suse/rmt/lib/rmt/cli/base.rb:40:in `handle_exceptions'
	 9: from /files/suse/rmt/lib/rmt/cli/base.rb:29:in `block in dispatch'
	 8: from /home/felixsch/.rvm/gems/ruby-2.5.8/gems/thor-1.1.0/lib/thor.rb:392:in `dispatch'
	 7: from /home/felixsch/.rvm/gems/ruby-2.5.8/gems/thor-1.1.0/lib/thor/invocation.rb:127:in `invoke_command'
	 6: from /home/felixsch/.rvm/gems/ruby-2.5.8/gems/thor-1.1.0/lib/thor/command.rb:27:in `run'
	 5: from /files/suse/rmt/lib/rmt/cli/main.rb:9:in `sync'
	 4: from /files/suse/rmt/lib/rmt/lockfile.rb:15:in `lock'
	 3: from /files/suse/rmt/lib/rmt/cli/main.rb:10:in `block in sync'
	 2: from /files/suse/rmt/lib/rmt/scc.rb:22:in `sync'
	 1: from /files/suse/rmt/lib/suse/connect/api.rb:40:in `list_products'
/files/suse/rmt/lib/suse/connect/api.rb:22:in `connect_api': Encountered an error validating htxxtp://xxxxxxlocalhost:3000. Be sure to add http/https if it's an absolute url, i.e IP Address (URI::InvalidURIError)

Maybe this can be handled in the CLI?
```

**Comment by ngetahun** [comment]
> lgtm 
> 
> **How I reviewed this pull request:**
> 
> * I run the branch and checked the default configuration works out of the box
> * Set the `scc » host` configuration to my localhost an check `sync` is working.
> * Set to a malformed string and see the exception.
> * Checked if there is no configuration at all fallback works
> 
> PS: Looks good, the only nitpicking thing is the exception is not handled and therefore the output is not really nicely formated:
> 
> ```
> I, [2021-09-22T11:05:41.825992 #6995]  INFO -- : Downloading data from SCC
> I, [2021-09-22T11:05:41.826055 #6995]  INFO -- : Updating products
> Traceback (most recent call last):
> 	13: from bin/rmt-cli:41:in `<main>'
> 	12: from /home/felixsch/.rvm/gems/ruby-2.5.8/gems/thor-1.1.0/lib/thor/base.rb:485:in `start'
> 	11: from /files/suse/rmt/lib/rmt/cli/base.rb:29:in `dispatch'
> 	10: from /files/suse/rmt/lib/rmt/cli/base.rb:40:in `handle_exceptions'
> 	 9: from /files/suse/rmt/lib/rmt/cli/base.rb:29:in `block in dispatch'
> 	 8: from /home/felixsch/.rvm/gems/ruby-2.5.8/gems/thor-1.1.0/lib/thor.rb:392:in `dispatch'
> 	 7: from /home/felixsch/.rvm/gems/ruby-2.5.8/gems/thor-1.1.0/lib/thor/invocation.rb:127:in `invoke_command'
> 	 6: from /home/felixsch/.rvm/gems/ruby-2.5.8/gems/thor-1.1.0/lib/thor/command.rb:27:in `run'
> 	 5: from /files/suse/rmt/lib/rmt/cli/main.rb:9:in `sync'
> 	 4: from /files/suse/rmt/lib/rmt/lockfile.rb:15:in `lock'
> 	 3: from /files/suse/rmt/lib/rmt/cli/main.rb:10:in `block in sync'
> 	 2: from /files/suse/rmt/lib/rmt/scc.rb:22:in `sync'
> 	 1: from /files/suse/rmt/lib/suse/connect/api.rb:40:in `list_products'
> /files/suse/rmt/lib/suse/connect/api.rb:22:in `connect_api': Encountered an error validating htxxtp://xxxxxxlocalhost:3000. Be sure to add http/https if it's an absolute url, i.e IP Address (URI::InvalidURIError)
> 
> Maybe this can be handled in the CLI?
> ```

I have made improvements, have a look!

**Comment by suse-tests-pass** [comment]
Well Done! Your tests are still passing.
https://ci.suse.de/job/scc-RMT-integration-tests/213992/console
If the given link has expired,you can force a Prophet rerun by just deleting this comment. (Merged 96da9b867de4786853a53df9fcd8146930c74e49 into d3226e0749bc27e5d424a626428115914e90d4d6)

**Comment by suse-tests-pass** [comment]
Well Done! Your tests are still passing.
https://ci.suse.de/job/scc-RMT-integration-tests/213993/console
If the given link has expired,you can force a Prophet rerun by just deleting this comment. (Merged d67ae1e72a9507a89d1b66c8b3836d305c775158 into d3226e0749bc27e5d424a626428115914e90d4d6)

**Comment by ngetahun** [comment]
Merge after prophet is done

**Comment by suse-tests-pass** [comment]
Well Done! Your tests are still passing.
https://ci.suse.de/job/scc-RMT-integration-tests/213996/console
If the given link has expired,you can force a Prophet rerun by just deleting this comment. (Merged 6a91541988f4c059b3a1b39175397829c4ee5968 into d3226e0749bc27e5d424a626428115914e90d4d6)

**Comment by suse-tests-pass** [comment]
Well Done! Your tests are still passing.
https://ci.suse.de/job/scc-RMT-integration-tests/213998/console
If the given link has expired,you can force a Prophet rerun by just deleting this comment. (Merged 6a91541988f4c059b3a1b39175397829c4ee5968 into 5d4e9ab6288e0df2be853d90c40387f21796de8b)

**Comment by suse-tests-pass** [comment]
Well Done! Your tests are still passing.
https://ci.suse.de/job/scc-RMT-integration-tests/214636/console
If the given link has expired,you can force a Prophet rerun by just deleting this comment. (Merged ea8965da8fdd75dd8288ca336d9b1407baed02a7 into 5d4e9ab6288e0df2be853d90c40387f21796de8b)

**Pull request merged**